test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1007 @@
+    PBSE_INCORRECT_USAGE = 2
+    fmt = "%a %b %d %H:%M:%S %Y"
+
+    def setUp(self):
+
+        TestFunctional.setUp(self)
+        # Set PBS_TZID, needed for standing reservation.
+        if 'PBS_TZID' in self.conf:
+            self.tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            self.tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Timezone not set, using Asia/Kolkata')
+            self.tzone = 'Asia/Kolkata'
+
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes('vnode', a, num=2, mom=self.mom)
+
+        rc = self.server.manager(MGR_CMD_SET, SERVER, {
+            'log_events': 4095}, expect=True)
+        self.assertTrue(rc)
+
+    def submit_and_confirm_reservation(self, offset, duration, standing=False,
+                                       select="1:ncpus=4",
+                                       rrule="FREQ=HOURLY;COUNT=2",
+                                       ExpectSuccess=1):
+        """
+        Helper function to submit a reservation and wait until it is confirmed.
+        It also checks for the corresponding server and accounting logs.
+
+        param offset: time in seconds after which the reservation should start.
+        type  offset: int
+
+        param duration: duration of the reservation to be submitted.
+        type  duration: int
+
+        param standing: whether to submit a standing reservation. Default: No.
+        type  standing: bool
+
+        param select: select specification for the reservation,
+                      Default: "1:ncpus=4"
+        type  select: string.
+
+        param rrule:  iCal recurrence rule for submitting a standing
+                      reservation.  Default: "FREQ=HOURLY;COUNT=2"
+        type  select: string.
+
+        param ExpectSuccess: Whether the caller expects the reservation
+                             to be successful or not.
+        type  ExpectSuccess: int.
+        """
+        start = int(time.time()) + offset
+        end = start + duration
+
+        if (standing):
+            attrs = {'Resource_List.select': select,
+                     'reserve_start': start,
+                     'reserve_end': end,
+                     'reserve_timezone': self.tzone,
+                     'reserve_rrule': rrule}
+        else:
+            attrs = {'Resource_List.select': select,
+                     'reserve_start': start,
+                     'reserve_end': end}
+
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+        msg = ".*Resv;" + rid + ";New reservation submitted start="
+        msg += time.strftime(self.fmt, time.localtime(int(start)))
+        msg += " end="
+        msg += time.strftime(self.fmt, time.localtime(int(end)))
+        if (standing):
+            msg += " recurrence_rrule=" + rrule + " timezone="
+            msg += self.tzone
+
+        self.server.log_match(msg, regexp=True, interval=2, max_attempts=30)
```

Comment:
some of the log_match need the regexp, I have retained those and removed the unwanted ones.